src/Adapter/Customer/QueryHandler/GetCustomerForViewingHandler.php

```diff
@@ -90,19 +98,30 @@ final class GetCustomerForViewingHandler implements GetCustomerForViewingHandler
      */
     private $link;
 
+    /**
+     * @var Locale
+     */
+    private $locale;
+
     /**
      * @param TranslatorInterface $translator
      * @param int $contextLangId
      * @param Link $link
+     * @param LocaleRepository $repository
      */
     public function __construct(
         TranslatorInterface $translator,
         $contextLangId,
-        Link $link
+        Link $link,
+        LocaleRepository $repository
     ) {
+        $this->context = new LegacyContext();
```

Comment:
Same comment as above, avoid injecting LegacyContext when you can

Reply:
:heavy_check_mark: